### DISCUSSION TITLE
github disables auto-merge automatically when requiring signed commits

### DISCUSSION THREAD

**Discussion by DavidS-ovm** [renovatebot/renovate#23054] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm trying to enable auto-merging for minor and patch releases. 

Our config: https://github.com/overmindtech/renovate-config/

A failing PR: https://github.com/overmindtech/aws-source/pull/227

![2023-06-30_09MS+0200_1256x511](https://github.com/renovatebot/renovate/assets/118179693/8fcda59c-be5e-477e-ab24-966a6a6749cc)

> auto-merge was automatically disabled 
> Base branch requires signed commits

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Ensuring PR (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: There are 0 errors and 0 warnings (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: getBranchPr(renovate/google.golang.org-protobuf-1.x) (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: findPr(renovate/google.golang.org-protobuf-1.x, undefined, open) (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: findPr(renovate/google.golang.org-protobuf-1.x, undefined, closed) (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: getPrCache() (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: Fetching changelog: https://github.com/protocolbuffers/protobuf-go (v1.30.0 -> v1.31.0) (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: Creating PR (branch="renovate/google.golang.org-protobuf-1.x")
{
  "prTitle": "Update module google.golang.org/protobuf to v1.31.0"
}

DEBUG: Creating PR (branch="renovate/google.golang.org-protobuf-1.x")
{
  "title": "Update module google.golang.org/protobuf to v1.31.0"
  "head": "overmindtech:renovate/google.golang.org-protobuf-1.x"
  "base": "main"
  "draft": false
}

DEBUG: PR created (branch="renovate/google.golang.org-protobuf-1.x")
{
  "pr": 227
  "draft": false
}

DEBUG: Adding labels 'dependencies' to #227 (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: GitHub-native automerge: success...PrNo: 227 (branch="renovate/google.golang.org-protobuf-1.x")
INFO: PR created (branch="renovate/google.golang.org-protobuf-1.x")
{
  "pr": 227
  "prTitle": "Update module google.golang.org/protobuf to v1.31.0"
}

DEBUG: resolveBranchStatus(branchName=renovate/google.golang.org-protobuf-1.x, ignoreTests=false) (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: getBranchStatus(renovate/google.golang.org-protobuf-1.x) (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: branch status check result (branch="renovate/google.golang.org-protobuf-1.x")
{
  "state": "pending"
  "statuses": []
}

DEBUG: Successful checks are all internal renovate/ checks, so returning "pending" branch status (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: No check runs found (branch="renovate/google.golang.org-protobuf-1.x")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}

DEBUG: Branch status yellow (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: Skipping assignees and reviewers as automerge=true (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: setPrCache() (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: Created Pull Request #227 (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: PR is configured for automerge (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: setBranchCommit() (branch="renovate/google.golang.org-protobuf-1.x")
DEBUG: getBranchPr(renovate/overmindtech-go)
DEBUG: findPr(renovate/overmindtech-go, undefined, open)
DEBUG: Found PR #225
DEBUG: getPrCache()
DEBUG: getBranchPr(renovate/aws-sdk-go)
DEBUG: findPr(renovate/aws-sdk-go, undefined, open)
DEBUG: Found PR #226
DEBUG: getPrCache()
DEBUG: getBranchPr(renovate/google.golang.org-protobuf-1.x)
DEBUG: findPr(renovate/google.golang.org-protobuf-1.x, undefined, open)
DEBUG: Found PR #227
[...]
DEBUG: Branch summary
{
  "cacheModified": true
  "baseBranches": [
    {
      "branchName": "main",
      "sha": "143426f6392f72f734318a7f12d654d5eea651e3"
    }
  ]
  "branches": [
    {
      "automerge": true,
      "baseBranch": "main",
      "baseBranchSha": "143426f6392f72f734318a7f12d654d5eea651e3",
      "branchName": "renovate/overmindtech-go",
      "branchSha": "8544edae038bc8bae8e9ca504eea61d6a0446226",
      "isModified": false,
      "isPristine": true
    },
    {
      "automerge": true,
      "baseBranch": "main",
      "baseBranchSha": "143426f6392f72f734318a7f12d654d5eea651e3",
      "branchName": "renovate/aws-sdk-go",
      "branchSha": "ce7862622a3949c02e0d484deac2ab6f0db29e27",
      "isModified": false,
      "isPristine": true
    },
    {
      "automerge": true,
      "baseBranch": "main",
      "baseBranchSha": "143426f6392f72f734318a7f12d654d5eea651e3",
      "branchName": "renovate/google.golang.org-protobuf-1.x",
      "branchSha": "e98a036b6f81143882966628285e9b6def881e37",
      "isModified": false,
      "isPristine": true
    }
  ]
  "defaultBranch": "main"
  "inactiveBranches": []
}

From https://developer.mend.io/github/overmindtech/aws-source/-/job/01890999-e61c-703c-9c1d-abae1d75a9cc
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Renovate's commits are signed. e.g. from the PR you linked to:

![image](https://github.com/renovatebot/renovate/assets/6311784/de49565a-5d86-4dc1-91c9-106f17c50a31)

It looks like GitHub has a bug or limitation that platform automerge doesn't work if signed commits are required?

**Reply by DavidS-ovm** [reply]
Yeah, that's what I'm not understanding. I was hoping someone on here would have had more experience with this than me.

**Reply by rarkins** [reply]
What type of automerge do you allow in the repo?

**Reply by DavidS-ovm** [reply]
Everything:
![2023-06-30_10MS+0200_1109x1172](https://github.com/renovatebot/renovate/assets/118179693/4b0d09fb-8e46-492b-a231-34367a1c645e)

**Reply by DavidS-ovm** [reply]
Here's also the branch protection rule of the repo: 
![2023-06-30_10MS+0200_938x1995](https://github.com/renovatebot/renovate/assets/118179693/829f6740-d4e5-4bf3-8802-e3fbeec35ffc)

**Reply by rarkins** [reply]
I looked in this repo and it seems Renovate's automerged commits aren't verified: 

![image](https://github.com/renovatebot/renovate/assets/6311784/6ba02720-af1a-4dfc-8e96-5fdb3a1a0b43)

I guess GitHub doesn't automatically sign/verify commits when automerging

**Reply by DavidS-ovm** [reply]
1) they are marked as verified in your screenshot?
2) we use auto merging enabled by humans regularly in the repos and it works just fine.

One thing I noticed is that this repo (renovate) uses a merge queue, whereas my repo doesn't have a merge queue and I see renovatebot enabling `auto-merge (rebase)`. Could that make a difference?

**Reply by DavidS-ovm** [reply]
Here's one I did just this morning:

![2023-06-30_10MS+0200_1376x1124](https://github.com/renovatebot/renovate/assets/118179693/c994346a-7b10-411a-adb2-e675fd5f6ab1)

and the resulting commit (squashed in this case):

![2023-06-30_10MS+0200_1663x144](https://github.com/renovatebot/renovate/assets/118179693/f97c0b97-dc7d-49d4-82be-e9a7537494f7)

**Reply by rarkins** [reply]
Yes, you're right - they're marked as Verified. I'm not sure how I got confused. So the mystery continues

**Reply by DavidS-ovm** [reply]
If you haven't changed anything, maybe github has gotten their act together. Either way, it works now:

![2023-07-07_08MS+0200_987x412](https://github.com/renovatebot/renovate/assets/118179693/8a6e700e-9bc4-47d7-968a-4b048ac1e22a)

**Reply by DavidS-ovm** [reply]
Only partially. Note how (for some unknown to me reason) on this repo the bot chooses rebase instead of squash:

![2023-07-07_08MS+0200_883x307](https://github.com/renovatebot/renovate/assets/118179693/6b5738d7-74e5-44f9-8a59-032b30ee77c3)

(from https://github.com/overmindtech/sdpcache/pull/52)

**Reply by rarkins** [reply]
You can change merge settings per-repo in github settings

**Reply by DavidS-ovm** [reply, marked as answer]
I'll try turning off rebase merging then. We don't really use that anyways.

**Reply by DavidS-ovm** [reply]
Yeah that worked. renovate happily automerges minor changes now as instructed.

**Comment by testworksau** [comment]
Hi @rarkins 👋🏼 

I came across this myself recently, and reached out to GitHub via our enterprise support on this; they responded with:

> It's possible to specify the default auto merge type using the enablePullRequestAutoMerge mutation with the mergeMethod input field:
> 
> ```
> mutation {
>     enablePullRequestAutoMerge(input: {pullRequestId: "$pull_request_id_graphql", mergeMethod: MERGE}) {
>         clientMutationId
>          }
> }
> ```
> 
> 
> The possible values for mergeMethod are:
> 
> `MERGE` - Add all commits from the head branch to the base branch with a merge commit.
> `SQUASH` - Combine all commits from the head branch into a single commit in the base branch.
> `REBASE` - Add all commits from the head branch onto the base branch individually.
> 
> Further information on usage can be found in [GitHub's Docs](https://docs.github.com/en/graphql/reference/input-objects#enablepullrequestautomergeinput).
